docs/examples/indie-linter-v2.md

```diff
@@ -0,0 +1,95 @@
+# Indie Linter v2
+
+The [Indie Linter API v2](../types/indie-linter-v2.md) demoed below supports [Messages v2](../types/linter-message-v2.md) only.
+
+## package.json
+
+Only the `consumedServices` field is important in this manifest, the rest are just for reference
+
+```json
+{
+  "name": "linter-example",
+  "main": "index.js",
+  "version": "0.0.0",
+  "private": true,
+  "description": "My Linter Package",
+  "keywords": ["linter-example", "example"],
+  "engines": {
+    "atom": ">=1.4.0 <2.0.0"
+  },
+  "consumedServices": {
+    "linter-indie": {
+      "versions": {
+        "2.0.0": "consumeIndie"
+      }
+    }
+  }
+}
+```
+
+## index.js
+
+```js
+'use babel'
+
+import { CompositeDisposable } from 'atom'
+
+let subscriptions
+export function activate() {
+  subscriptions = new CompositeDisposable()
+}
+export function deactivate() {
+  subscriptions.dispose()
+}
+export function consumeIndie(registerIndie) {
+  const linter = registerIndie({
+    name: 'My Linter',
+  })
+  subscriptions.add(linter)
+
+  // Example using ::setMessages
+  subscriptions.add(atom.workspace.observeTextEditors(function(textEditor) {
+    const editorPath = textEditor.getPath()
+    if (!editorPath) {
+      return
+    }
+
+    linter.setMessages(editorPath, [{
+      severity: 'info',
+      location: {
+        path: editorPath,
+        position: [[0, 0], [0, Infinity]],
```

Comment:
Maybe put a comment explaining the `Infinity` here for those that haven't had to dig into the Atom source code to know what in the world that does 😛.